linux/setup_repo.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# This script installs a Jamulus repository to Debian based systems
+
+if [[ ${EUID} -ne 0 ]]; then
+     echo "Error: This script must be run as root."
+     exit 1
+fi
+
+REPO_FILE=/etc/apt/sources.list.d/jamulus.list
+KEY_FILE=/etc/apt/trusted.gpg.d/jamulus.asc
+GITHUB_REPOSITORY="jamulussoftware/jamulus"
+
+echo "Setting up Jamulus repo at ${REPO_FILE}..."
+echo "deb https://github.com/${GITHUB_REPOSITORY}/releases/latest/download/ ./" > ${REPO_FILE}
+echo "Installing Jamulus GPG key at ${KEY_FILE}..."
+curl --fail --show-error -sLo "${KEY_FILE}" https://github.com/${GITHUB_REPOSITORY}/releases/latest/download/key.asc
```

Comment:
Just trying this script locally, and this step fails with a 404 for `key.asc`.

Is this script intended for use by a Github workflow, or by an end user? I can't see it referred to anywhere else in the tree.

Reply:
> Is this script intended for use by a Github workflow, or by an end user? I can't see it referred to anywhere else in the tree.

Purely by our git build to create a repo for apt to auto update from.

Reply:
You'd need to replace the variables. The release will generate a runnable script.

Reply:
I've currently checked out `ann0see/debrepo`, but I can't see anywhere that calls `setup_repo.sh`. I would have expected it to be referred to somewhere within `autobuild.yml`? Or within another script somewhere in the tree?

You can tell I'm a bit rusty!

Reply:
setup_repo.sh is a script which the user should run on his machine to install the repository. The logic to create the repository is in autobuild.yml. I believe the content in autobuild.yaml correctly replaces the content of the setup_repo.sh script and uploads the modified version to the release.